plugins/modules/mso_schema_template_contract_filter.py

```diff
@@ -187,14 +187,16 @@ def main():
     )
 
     schema = module.params.get('schema')
-    template = module.params.get('template')
+    template = module.params.get('template').replace(' ', '')
     contract = module.params.get('contract')
     contract_display_name = module.params.get('contract_display_name')
     contract_filter_type = module.params.get('contract_filter_type')
     contract_scope = module.params.get('contract_scope')
     filter_name = module.params.get('filter')
     filter_directives = module.params.get('filter_directives')
     filter_template = module.params.get('filter_template')
+    if filter_template is not None:
+        filter_template = filter_template.replace(' ', '')
```

Comment:
I would put the replace on line 197 like the other occurrences

Reply:
The template of filter is not required, so we need to check whether the template is none before replacing.